app/src/main/java/com/spencerpages/MainActivity.java

```diff
@@ -613,27 +616,20 @@ private void handleImportCollectionsPart1(){
             // characters in the export file names (which the OS interprets as directory delimiters)
             String collectionFileName = collectionData[0].replaceAll("/", "_SL_");
 
-    		file = new File(dir, collectionFileName + ".csv");
+    		inputFile = new File(dir, collectionFileName + ".csv");
 
-    		if(!file.isFile()){
-    			showCancelableAlert(mRes.getString(R.string.cannot_find_input_file, file.getAbsolutePath()));
-    			return;
+    		if(!inputFile.isFile()){
+            	showCancelableAlert(mRes.getString(R.string.cannot_find_input_file, inputFile.getAbsolutePath()));
+            	return;
     		}
 
-    		try {
-    			f = new FileInputStream(file);
-    			in_ = new InputStreamReader(f, "UTF8");
-    			in = new BufferedReader(in_);
-    		} catch (Exception e) {
-    			showCancelableAlert(mRes.getString(R.string.error_open_file_reading, file.getAbsolutePath()));
-    			return;
-    		}
+    		in = openFileForReading(inputFile);
+    		if(in == null) return;
 
     		ArrayList<String[]> collectionContent = new ArrayList<>();
 
     		try {
-    			String line = in.readLine();
```

Comment:
Ah, I see why line's scope was increased.  I could go either way, although I think I'd prefer having two declarations of line that are both more narrowly scoped.  Thoughts?